docs/en/reference/working-with-associations.rst

```diff
@@ -437,8 +443,10 @@ only accessing it through the User entity:
     // User entity
     class User
     {
-        private $id;
-        private $comments;
+        private int $id;
+
+        /** @var Collection<int, Comment> */
```

Comment:
Here: key can be mixed as we don't know how people are adding to collection via add or set :)

Reply:
And if they do, they should get a warning from PHPStan/Psalm thanks to this annotation, so I stand by it.